iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_queue_pusher.hpp

```diff
@@ -11,12 +12,18 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
 #ifndef IOX_POSH_POPO_BUILDING_BLOCKS_CHUNK_QUEUE_PUSHER_HPP
 #define IOX_POSH_POPO_BUILDING_BLOCKS_CHUNK_QUEUE_PUSHER_HPP
 
+#include "iceoryx_posh/internal/log/posh_logging.hpp"
 #include "iceoryx_posh/internal/mepoo/shared_chunk.hpp"
 #include "iceoryx_posh/internal/popo/building_blocks/chunk_queue_data.hpp"
 #include "iceoryx_posh/internal/popo/building_blocks/chunk_queue_types.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/condition_variable_signaler.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/event_notifier.hpp"
```

Comment:
This is very bad style. The `inl` file is an extension of the header file, so either way the include will be parsed by the compiler. But when we put it in the `inl` file we hide that information from every one. The sole purpose of an inline file is to implement the inline code and not to add includes - this is the purpose of the header to provide the headers required for the definition and inline code implementation.

Reply:
I don't see why this is bad style. With `inl` files we want to provide a clear separation between definition and declaration. IMHO this also includes the dependencies of defintion and declaration. You are right, that it doesn't give as a benefit compile time wise here, but think of the case that e.g. with the next refactoring the template would become a normal class. Then you need those two includes in the `cpp` and not in the `hpp`. With the includes being right where they belong in the first place this will become easier.

Reply:
It can be done like with hpp and cpp but has much less benefit.

As you stated the compiler sees all of these and the dependencies are still there in the hpp that includes the inl. The question is: does the user need to know? I prefer having no includes in the inl file myself, except for the hpp that is implemented ( only because my IDE goes nuts as it cannot see the definitions, can maybe solved otherwise).

TLDR:In a hpp/inl split I support having everything in the hpp for transparency.
It helps to see the inl file just as something that follows the definitions. We would not want includes there if it was all in one file...